### PR TITLE
chore: Separate semver

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
## Description

Closes: https://github.com/cosmos/interchain-security/issues/1216
---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

**Review by p-offtermatt — APPROVED** [review]
(Whether this ends up being the way we want to handle the separate versions is a separate issue, just commenting on the content of this PR.)

The PR looks good. It might be worth more explicitly writing how to tell whether two releases are mutually compatible.
I think all releases with the same major version should be compatible?

Before we merge this PR, we should check whether there are missing references to the versioning scheme that should be adjusted.
I checked the docs website and didn't find anything that needs adjusting, and no other place came to mind.

File: .github/PULL_REQUEST_TEMPLATE/production.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ I have...
 * [ ] Updated the relevant documentation or specification
 * [ ] Reviewed "Files changed" and left comments if necessary <!-- relevant if the changes are not obvious  -->
 * [ ] Confirmed all CI checks have passed
+* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release for both the consumer and provider
```
Comment (mpoke) [inline .github/PULL_REQUEST_TEMPLATE/production.md:33]: Do we want to do this for every API breaking PR?

**Comment by smarshall-spitzbart** [comment]
> It might be worth more explicitly writing how to tell whether two releases are mutually compatible. I think all releases with the same major version should be compatible?

See section on [backwards compatibility ](https://github.com/cosmos/interchain-security/blob/e9c178eb1a444308dc1866ea12e90cb7fe8d23d2/CONTRIBUTING.md#backwards-compatibility) which will remain unchanged by this PR

File: .github/PULL_REQUEST_TEMPLATE/production.md (hunk 1 above)
Comment (smarshall-spitzbart) [reply .github/PULL_REQUEST_TEMPLATE/production.md:33]: If we bump the major version of ICS, golang requires us to bump the version string. We can redefine what constitutes a major release, but that'd be steering away from how IBC does versioning

Comment (mpoke) [reply .github/PULL_REQUEST_TEMPLATE/production.md:33]: My comment is more generally about the mapping between a single PR and a release. Yes, a release that contains a PR with library API breaking changes should have the major version bumped. But not every PR with library API breaking changes should result in a new release.

**Review by mpoke — APPROVED** [review]

**Pull request merged**